src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -98,6 +102,11 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                     functionName = string.IsNullOrEmpty(fex.MethodName) ? string.Empty : fex.MethodName.Replace("Host.Functions.", string.Empty);
                 }
 
+                if (exception is FunctionIndexingException && _eventManager != null)
+                {
+                    _eventManager.Publish(new FunctionIndexingEvent(eventName, source, exception));
```

Comment:
I think it might be better to put propagation actions like this at the top of the method before any conditional filtering is applied, to ensure these are never filtered out. Add a comment line "propagate special exceptions through the event bus"